src/server/trigger.rs

```diff
@@ -105,6 +105,21 @@ impl<T> Inner<T> {
         condvar.notify_one();
     }
 
+    fn disconnect(&self) {
+        let &Inner { state_mutex, condvar: _ } = &self;
+        let state_guard = state_mutex.lock().expect(POISON_PANIC);
+
+        let disconnect_value = if let State::Activated(value) = &*state_guard {
+            let read_value: T = unsafe { ptr::read(value) };
+            Some(read_value)
+        } else {
+            None
+        };
```

Comment:
```suggestion
        let state = std::mem::replace(&mut *state_guard, State::Pending);

        let disconnect_value = if let State::Activated(value) = state { Some(value) } else { None };
```

We can replace it with dummy value, so there will be no need for unsafe, up 2 you